### PR TITLE
APPDEV-1246 Friends dashboard navigation title header issue

### PR DESCRIPTION
**Pull request by vishal-revdiwala**
- Code updated to show buddy first name and last name in header
- Profile icon of dashboard, display buddy pic if uploaded, otherwise first letter of nick name
- Handled buddy pic inside friend profile view
- Code refactor

### CODE REVIEW

File: Yona/Yona/CustomCells/YonaUserHeaderWithTwoTabTableViewCell.swift [hunk 1]
```diff
@@ -135,20 +135,19 @@ class YonaUserHeaderWithTwoTabTableViewCell: UITableViewCell {
...
+        if let nickName = aBuddy?.buddyNickName {
+            nicknameLabel.text = nickName
         }
-        nameLabel.text = "\(tmpFirst) \(tmpLast)"
-        if let txt = aBuddy?.buddyNickName {
-            nicknameLabel.text = txt
+        if let link = buddy.buddyAvatarURL, let URL = URL(string: link) {
+            avatarImageView.kf.setImage(with: URL)
+        } else {
+            if buddy.UserRequestfirstName.count > 0 && buddy.UserRequestlastName.count > 0{
```
Comment (Bert-R) [inline Yona/Yona/CustomCells/YonaUserHeaderWithTwoTabTableViewCell.swift]: Do you have scenarios where the first or last name is empty? I don't think that ever occurs. If there are no realistic use cases, remove the condition. If it has to stay, change it this way:
```
        } else if buddy.UserRequestfirstName.count > 0 && buddy.UserRequestlastName.count > 0 {
```

File: Yona/Yona/CustomCells/YonaUserHeaderWithTwoTabTableViewCell.swift [hunk 2]
```diff
@@ -135,20 +135,19 @@ class YonaUserHeaderWithTwoTabTableViewCell: UITableViewCell {
...
+                avatraInitialsLabel.text =  "\(buddy.UserRequestfirstName.capitalized.first!) \(buddy.UserRequestlastName.capitalized.first!)"
```
Comment (Bert-R) [inline Yona/Yona/CustomCells/YonaUserHeaderWithTwoTabTableViewCell.swift]: Please rename ``avatraInitialsLabel`` into ``avatarInitialsLabel``

File: Yona/Yona/Friends/FriendsDashboardViewController.swift [hunk 3]
```diff
@@ -71,23 +71,30 @@ class FriendsDashboardViewController: MeDashBoardMainViewController {
...
-                btnName.layer.cornerRadius = btnName.frame.size.width/2
-                btnName.layer.borderWidth = 1
-                btnName.layer.borderColor = UIColor.white.cgColor
-                let rightBarButton = UIBarButtonItem()
-                rightBarButton.customView = btnName
-                navigationItem.rightBarButtonItems = [rightBarButton]
+                btnName.clipsToBounds = true
+            })
+        } else if let nickName = buddyToShow?.buddyNickName {
+            if nickName.count > 0 {
```
Comment (Bert-R) [inline Yona/Yona/Friends/FriendsDashboardViewController.swift]: To the best of my knowledge, the nickname of a buddy is always set. If you agree there are no realistic use cases where it is missing or empty, the remove the conditions.

File: Yona/Yona/CustomCells/YonaUserHeaderWithTwoTabTableViewCell.swift (hunk 1 above)
Comment (vishal-revdiwala) [reply Yona/Yona/CustomCells/YonaUserHeaderWithTwoTabTableViewCell.swift]: Yes, first name and last name is always set. Checks are not required. 
Code updated accordingly.

File: Yona/Yona/CustomCells/YonaUserHeaderWithTwoTabTableViewCell.swift (hunk 2 above)
Comment (vishal-revdiwala) [reply Yona/Yona/CustomCells/YonaUserHeaderWithTwoTabTableViewCell.swift]: Done

File: Yona/Yona/Friends/FriendsDashboardViewController.swift (hunk 3 above)
Comment (vishal-revdiwala) [reply Yona/Yona/Friends/FriendsDashboardViewController.swift]: I agree, there are no such cases where nick name is empty. Removed one extra if check.

**Pull request merged**